#VRML_SIM R2020b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9886202756980692 -0.14498250836405066 -0.04012508874933063 0.5459132120711018
  position -0.784446670966857 1.684568948689861 3.005678322878047
}
Background {
  skyColor [
    0.15 0.45 1
  ]
}
Floor {
}
Robot {
  translation 0 0.5 0
  children [
    InertialUnit {
    }
    Gyro {
    }
    DEF FL Propeller {
      shaftAxis 0 1 0
      centerOfThrust 0.1 0 -0.1
      device RotationalMotor {
        name "front left propeller"
      }
      fastHelix Solid {
        translation 0.1 0 -0.1
        children [
          DEF PROPELLER Shape {
            appearance DEF FAST_HELIX_APPEARANCE PBRAppearance {
              transparency 0.5
              roughness 0.5
              metalness 0
            }
            geometry Cylinder {
              height 0.003
              radius 0.145
              side FALSE
              subdivision 24
            }
          }
        ]
      }
      slowHelix Solid {
        translation 0.1 0 -0.1
        children [
          DEF SLOW_TAIL_HELIX Transform {
            rotation 1 0 0 1.5708
            children [
              DEF SLOW_TAIL_HELIX_GROUP Group {
                children [
                  DEF HUB Shape {
                    appearance DEF BLACK_METAL PBRAppearance {
                      baseColor 0 0 0
                      roughness 1.1102230246251565e-16
                      metalness 0
                    }
                    geometry Capsule {
                      height 0.01
                      radius 0.01
                    }
                  }
                  DEF BLADE_PITCH_AXIS Shape {
                    appearance USE BLACK_METAL
                    geometry Cylinder {
                      height 0.05
                      radius 0.004
                    }
                  }
                  DEF RIGHT_BLADE Transform {
                    translation 0 0.04 0
                    rotation 0 1 0 0.3618
                    children [
                      DEF BLADE_SHAPE Shape {
                        appearance PBRAppearance {
                          baseColor 0.8 0.8 0.8
                          roughness 1
                          metalness 0
                        }
                        geometry Box {
                          size 0.006 0.03 0.002
                        }
                      }
                    ]
                  }
                  DEF RIGHT_BLADE Transform {
                    translation 0 -0.04 0
                    rotation 0 1 0 -0.3618
                    children [
                      USE BLADE_SHAPE
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    }
    DEF FR Propeller {
      shaftAxis 0 1 0
      centerOfThrust 0.1 0 0.1
      device RotationalMotor {
        name "front right propeller"
      }
      fastHelix Solid {
        translation 0.1 0 0.1
        children [
          USE PROPELLER
        ]
      }
      slowHelix Solid {
        translation 0.1 0 0.1
        children [
          DEF SLOW_TAIL_HELIX Transform {
            rotation 1 0 0 1.5708
            children [
              DEF SLOW_TAIL_HELIX_GROUP Group {
                children [
                  DEF HUB Shape {
                    appearance DEF BLACK_METAL PBRAppearance {
                      baseColor 0 0 0
                      roughness 1.1102230246251565e-16
                      metalness 0
                    }
                    geometry Capsule {
                      height 0.01
                      radius 0.01
                    }
                  }
                  DEF BLADE_PITCH_AXIS Shape {
                    appearance USE BLACK_METAL
                    geometry Cylinder {
                      height 0.05
                      radius 0.004
                    }
                  }
                  DEF RIGHT_BLADE Transform {
                    translation 0 0.04 0
                    rotation 0 1 0 0.3618
                    children [
                      DEF BLADE_SHAPE Shape {
                        appearance PBRAppearance {
                          baseColor 0.8 0.8 0.8
                          roughness 1
                          metalness 0
                        }
                        geometry Box {
                          size 0.006 0.03 0.002
                        }
                      }
                    ]
                  }
                  DEF RIGHT_BLADE Transform {
                    translation 0 -0.04 0
                    rotation 0 1 0 -0.3618
                    children [
                      USE BLADE_SHAPE
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    }
    DEF RL Propeller {
      shaftAxis 0 1 0
      centerOfThrust -0.1 0 -0.1
      device RotationalMotor {
        name "rear left propeller"
      }
      fastHelix Solid {
        translation -0.1 0 -0.1
        children [
          USE PROPELLER
        ]
      }
      slowHelix Solid {
        translation -0.1 0 -0.1
        children [
          DEF SLOW_TAIL_HELIX Transform {
            rotation 1 0 0 1.5708
            children [
              DEF SLOW_TAIL_HELIX_GROUP Group {
                children [
                  DEF HUB Shape {
                    appearance DEF BLACK_METAL PBRAppearance {
                      baseColor 0 0 0
                      roughness 1.1102230246251565e-16
                      metalness 0
                    }
                    geometry Capsule {
                      height 0.01
                      radius 0.01
                    }
                  }
                  DEF BLADE_PITCH_AXIS Shape {
                    appearance USE BLACK_METAL
                    geometry Cylinder {
                      height 0.05
                      radius 0.004
                    }
                  }
                  DEF RIGHT_BLADE Transform {
                    translation 0 0.04 0
                    rotation 0 1 0 0.3618
                    children [
                      DEF BLADE_SHAPE Shape {
                        appearance PBRAppearance {
                          baseColor 0.8 0.8 0.8
                          roughness 1
                          metalness 0
                        }
                        geometry Box {
                          size 0.006 0.03 0.002
                        }
                      }
                    ]
                  }
                  DEF RIGHT_BLADE Transform {
                    translation 0 -0.04 0
                    rotation 0 1 0 -0.3618
                    children [
                      USE BLADE_SHAPE
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    }
    DEF RR Propeller {
      shaftAxis 0 1 0
      centerOfThrust -0.1 0 0.1
      device RotationalMotor {
        name "rear right propeller"
      }
      fastHelix Solid {
        translation -0.1 0 0.1
        children [
          USE PROPELLER
        ]
      }
      slowHelix Solid {
        translation -0.1 0 0.1
        children [
          DEF SLOW_TAIL_HELIX Transform {
            rotation 1 0 0 1.5708
            children [
              DEF SLOW_TAIL_HELIX_GROUP Group {
                children [
                  DEF HUB Shape {
                    appearance DEF BLACK_METAL PBRAppearance {
                      baseColor 0 0 0
                      roughness 1.1102230246251565e-16
                      metalness 0
                    }
                    geometry Capsule {
                      height 0.01
                      radius 0.01
                    }
                  }
                  DEF BLADE_PITCH_AXIS Shape {
                    appearance USE BLACK_METAL
                    geometry Cylinder {
                      height 0.05
                      radius 0.004
                    }
                  }
                  DEF RIGHT_BLADE Transform {
                    translation 0 0.04 0
                    rotation 0 1 0 0.3618
                    children [
                      DEF BLADE_SHAPE Shape {
                        appearance PBRAppearance {
                          baseColor 0.8 0.8 0.8
                          roughness 1
                          metalness 0
                        }
                        geometry Box {
                          size 0.006 0.03 0.002
                        }
                      }
                    ]
                  }
                  DEF RIGHT_BLADE Transform {
                    translation 0 -0.04 0
                    rotation 0 1 0 -0.3618
                    children [
                      USE BLADE_SHAPE
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Box {
        size 0.1 0.01 0.1
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "newdrone"
}
